### ISSUE TITLE
Add a debug panel, containing sections with relevant debugging information

### ISSUE THREAD

**Issue by filipdutescu** [helix-editor/helix#5954]
# Description

Currently, it is very hard to view the values of variables and their inner fields, see all existing breakpoints, the call stack and there are no existing watches. This makes it very hard to debug any complex program, where any or all of the aforementioned items would be necessary.

# Proposal

Add a configurable panel of widgets for locals, watches, call stack and breakpoints, which should be able to be placed in any of the following positions: left, top, bottom, right. Each widget should itself be placeable independently from the others, in any of those directions. Those settings should be set in the `config.toml` file.

Values should be expandable, using the same mechanism that the explorer will expand directories (see #5768). They and watches should also be editable, with a prompt shown beneath the statusline, prefilled with the existing value.

# UI mockups

![image](https://user-images.githubusercontent.com/46979062/218332136-19d364e1-302e-4b11-849a-c3e698b92387.png)

**Comment by CptPotato** [comment]
Looks really nice. I'm personally a fan of "less borders", so here's some variations:

<details>
<summary>screenshot</summary>

![image](https://user-images.githubusercontent.com/3957610/218457449-e7234204-7716-40b9-8c63-f1acfef9f60f.png)

If the debug panel uses a different background, the vertical divider can be left out, too:

![image](https://user-images.githubusercontent.com/3957610/218458485-75fcdfb8-c23b-447e-9015-07d138d87651.png)
</details>

**Comment by filipdutescu** [comment]
Thanks a lot for the feedback and kind words! I do really like your suggestions, especially the one with only the top separator, at the right of the title.

I did it the way I did since there is already support for that exact border right now in Helix. I thought it would make this issue even more complex to allow customisation of borders for each widget. Is this assessment correct? Ideally you would be able to choose from all styles

**Comment by CptPotato** [comment]
> I did it the way I did since there is already support for that exact border right now in Helix. I thought it would make this issue even more complex to allow customisation of borders for each widget. Is this assessment correct? Ideally you would be able to choose from all styles

I guess you're right. The style of border you chose is in line with the existing ones, so maybe it's better to stay consistent for the first attempt.

**Comment by pascalkuthe** [comment]
I doubt it would actually be much harder to render this way, Rendering border like this is mostly custom code and not that complex to begin with. I also prefer the variants suggested by @CptPotato as that makes the screen less cluttered. I don't think we necesssairly need to offer customization here and instead just choose one good default look

**Comment by filipdutescu** [comment]
> I guess you're right. The style of border you chose is in line with the existing ones, so maybe it's better to stay consistent for the first attempt.

If you are so kind, you could open an issue precisely for this improvement and add it to the tracking issue

**Comment by pascalkuthe** [comment]
> > I guess you're right. The style of border you chose is in line with the existing ones, so maybe it's better to stay consistent for the first attempt.
> 
> If you are so kind, you could open an issue precisely for this improvement and add it to the tracking issue

 I don't think this is a separate issue and just a detail of how the panel would be styled

**Comment by pascalkuthe** [comment]
> > I doubt it would actually be much harder to render this way, Rendering border like this is mostly custom code and not that complex to begin with. I also prefer the variants suggested by @CptPotato as that makes the screen less cluttered. I don't think we necesssairly need to offer customization here and instead just choose one good default look
> 
> Then I would go for the first option @CptPotato presented, which can have the divider hidden as well, seems cleanest

In that case it might be nice to update the issue description with the new mockup

**Comment by filipdutescu** [comment]
Will do later. I will update the tracking one as well, so there is a history of how the design changed (this way we might get even more feedback).
